Cql/Cql.Compiler/ExpressionBuilder.cs

```diff
@@ -1034,33 +1036,11 @@ protected Expression FunctionRef(FunctionRef op)
                 .Concat(new[] { functionType })
                 .ToArray();
 
-            var funcType = GetFuncType(funcTypeParameters);
-
-            // FHIRHelpers has special handling in CQL-to-ELM and does not translate correctly - specifically,
-            // it interprets ToString(value string) oddly.  Normally when string is used in CQL it is resolved to the elm type.
-            // In FHIRHelpers, this string gets treated as a FHIR string, which is normally mapped to a StringElement abstraction.
-            if (op.libraryName != null && op.libraryName.StartsWith("fhirhelpers", StringComparison.OrdinalIgnoreCase))

```

Comment:
It could well be that we still need this to be here, as it seems to suggest the ToString() function in FHIRHelpers is incorrect (for our purposes)

Reply:
Although I cannot follow Evan's wording here, he's right that there's something odd about the ToString() function - it uses the `.value` property, which does not really exist in FhirPath, to refer to the atomic value of a primitive (not the primitive itself)  (BTW it even does that for non-primitives). This `value` property requires special handling in the model, so the actual `ToString` function cannot currently be used or called. So we need to reroute the call to the convert to string. Since this ultimately calls the model-specific TypeConverter, it gives the model the opportunity to do model-specific mapping of the `value` property, which is exactly what we want.